MSAL/src/native_auth/public/state_machine/delegate/SignUpDelegateDispatchers.swift

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+final class SignUpPasswordStartDelegateDispatcher: DelegateDispatcher<SignUpPasswordStartDelegate> {
+
+    func dispatchSignUpCodeRequired(
+        newState: SignUpCodeRequiredState,
+        sentTo: String,
+        channelTargetType: MSALNativeAuthChannelType,
+        codeLength: Int
+    ) async {
+        if let signUpCodeRequired = delegate.onSignUpCodeRequired {
+            telemetryUpdate?(.success(()))
+            await signUpCodeRequired(newState, sentTo, channelTargetType, codeLength)
+        } else {
+            let error = SignUpPasswordStartError(type: .generalError, message: errorMessage(for: "onSignUpCodeRequired"))
+            telemetryUpdate?(.failure(error))
+            await delegate.onSignUpPasswordError(error: error)
+        }
+    }
+
+    func dispatchSignUpAttributesInvalid(attributeNames: [String]) async {
+        if let signUpAttributes = delegate.onSignUpAttributesInvalid {
```

Comment:
This is a worse mismatch of the variable name <> function  mapping that I mentioned in another comment. I think this variable should be called `onSignUpAttributesInvalid` for consistency.

Reply:
Done (same as [here](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1914#discussion_r1400964541))